jira_agile_metrics/calculators/ageingwip.py

```diff
@@ -24,19 +24,9 @@ def run(self, today=None):
         cycle_data = self.get_result(CycleTimeCalculator)
         cycle_names = [s['name'] for s in self.settings['cycle']]
 
-        start_column = self.settings['committed_column']
-        end_column = self.settings['final_column']
+        committed_column = self.settings['committed_column']
         done_column = self.settings['done_column']
-        
-        if start_column not in cycle_names:
-            logger.error("Committed column %s does not exist", start_column)
-            return None
-        if end_column not in cycle_names:
-            logger.error("Final column %s does not exist", end_column)
-            return None
-        if done_column not in cycle_names:
-            logger.error("Done column %s does not exist", done_column)
-            return None
+        last_active_column = cycle_names[cycle_names.index(done_column)-1]
```

Comment:
The column described by the `end_column` config parameter is meant to allow the user to set this as a column that might not be the second-to-last-one (although it defaults to that, see https://github.com/DeloitteDigitalUK/jira-agile-metrics/blob/master/jira_agile_metrics/config.py#L477).

This is helpful because it allows you to define a "wide" cycle (e.g. in a file extended by multiple configurations) but also run certain analysis over a subset (we have a shared JIRA workflow that goes from very early analysis to live, but some of our analytics run on a subset of this workflow, e.g. a "dev team" flow).

Is there a reason it has to be hardcoded like this?

Reply:
As described in https://github.com/DeloitteDigitalUK/jira-agile-metrics/pull/27#issuecomment-636308642 I'd argue, that one can achieve the same effect with `Committed column` and `Done column`, that is by leaving the `Workflow` the same and just chaning `Committed column` and `Done column`.